app/validators/user_validator.rb

```diff
@@ -0,0 +1,7 @@
+class UserValidator < ActiveModel::Validator
+  def validate(record)
+    if User.find_by_email(record.email)
```

Comment:
I think we can use `exists?` for a slightly cheaper query (doesn't create a full AR object, doesn't load all columns frmo the DB)

```ruby
User.where(email: record.email).exists?
```

Reply:
Oh - this is good to know!  Thank you.